src/store.ts

```diff
@@ -97,7 +156,18 @@ const store: TypedVuexStore = new TypedVuexStore({
     uniqueIncrementer: 0,
     isTeleportModalOpen: false,
   },
-  actions: {},
+  actions: {
+    setSemesters(context, semester: readonly FirestoreSemester[]) {
+      const newSemesters = [];
+      for (let i = 0; i < semester.length; i += 1) {
+        const curSemester = semester[i];
+        newSemesters.push(popSemCourseData(curSemester));
```

Comment:
Hm, `popSemCourseData` returns a `Promise<FirestoreSemester>` right? We should probably `await` this so we don't end up with an array of promises. Specifically, right now `newSemesters` has type `Promise<FirestoreSemester>[]`, but we want it to have type `FirestoreSemester[]`.